manager/logbroker/broker.go

```diff
@@ -139,10 +139,14 @@ func (lb *LogBroker) registerSubscription(subscription *subscription) {
 	lb.registeredSubscriptions[subscription.message.ID] = subscription
 	lb.subscriptionQueue.Publish(subscription)
 
-	// Mark nodes that won't receive the message as done.
 	for _, node := range subscription.Nodes() {
 		if _, ok := lb.connectedNodes[node]; !ok {
-			subscription.Done(node, fmt.Errorf("node %s is not available", node))
+			// Mark nodes that won't receive the message as done.
+			lb.markDone(subscription, node, fmt.Errorf("node %s is not available", node))
+		} else {
+			// otherwise, add the subscription to the node's subscriptions list
+			log.L.WithField("area", "registerSubscription").Debugf("add subscription to node %v", node)
+			lb.connectedNodes[node][subscription] = struct{}{}
```

Comment:
Out of an abundance of caution, let's make sure this is not overwriting an existing subscription in `connectedNodes`.